Sources/SwiftGenKit/Parsers/Parser.swift

```diff
@@ -23,9 +25,23 @@ public protocol Parser {
 }
 
 public extension Parser {
-  func parse(paths: [Path]) throws {
+  func searchAndParse(paths: [Path], filter: Filter) throws {
     for path in paths {
-      try parse(path: path)
+      try searchAndParse(path: path, filter: filter)
+    }
+  }
+
+  func searchAndParse(path: Path, filter: Filter) throws {
+    if path.matches(filter: filter) {
+      let parentDir = path.absolute().parent()
+      try parse(path: path, relativeTo: parentDir)
+    } else {
+      let dirChildren = path.iterateChildren(options: [.skipsHiddenFiles, .skipsPackageDescendants])
+      let parentDir = path.absolute()
+
+      for path in dirChildren where path.matches(filter: filter) {
+        try parse(path: path, relativeTo: parentDir)
```

Comment:
In which case can `relativeTo:` not be the parent of the file passed to `path:`? I mean, if it's always the parent, then why pass it as a parameter instead of having the implementations of `parse(path:)` which  might need to know the parent be the one calling `path.absolute().parent()` to get it?

Reply:
Take the fonts case. Let's say I `searchAndParse` my `Resources/` folder. My fonts are located inside `Resources/Stuff/Fonts/`, and I'm copying those fonts into my app at that specific path.

The parser would need to preserve paths relative to the `Resources` folder, not the parent of `Fonts` (which'd be `Stuff`).